packages/graphql/tests/integration/rfcs/query-limits.int.test.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Driver } from "neo4j-driver";
+import { graphql } from "graphql";
+import { generate } from "randomstring";
+import neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src/classes";
+import { generateUniqueType } from "../../utils/graphql-types";
+
+describe("integration/rfcs/query-limits", () => {
+    let driver: Driver;
+
+    beforeAll(async () => {
+        driver = await neo4j();
+    });
+
+    afterAll(async () => {
+        await driver.close();
+    });
+
+    describe("Top Level Query Limits", () => {
+        test("should limit the top level query", async () => {
+            const session = driver.session();
+            const randomType = generateUniqueType("Movie");
+
+            const typeDefs = `
+                type ${randomType.name} @queryOptions(defaultLimit: 2) {
+                    id: ID!
+                }
+            `;
+
+            const neoSchema = new Neo4jGraphQL({ typeDefs });
+
+            try {
+                await session.run(
+                    `
+                        WITH [1,2,3,4,5] AS iterate
+                        UNWIND iterate AS i
+                        CREATE (:${randomType.name} {id: randomUUID()})
+                    `,
+                    {}
+                );
+
+                const query = `
+                        {
+                            ${randomType.plural} {
+                                id
+                            }
+                        }
+                `;
+
+                const gqlResult = await graphql({
+                    schema: neoSchema.schema,
+                    source: query,
+                    contextValue: { driver, driverConfig: { bookmarks: session.lastBookmark() } },
+                });
+
+                if (gqlResult.errors) {
+                    console.log(JSON.stringify(gqlResult.errors, null, 2));
+                }
+
+                expect(gqlResult.errors).toBeUndefined();
+                expect((gqlResult.data as any)[randomType.plural]).toHaveLength(2);
+            } finally {
+                await session.close();
```

Comment:
**question/suggestion (non-blocking)**: Is there a reason for not using the `beforeEach` and `afterEach` hooks for this?

```typescript
let session: Session

beforeEach(()=>{
  session = driver.session();
});

afterEach(async()=>{
  await session.close();
});
```

The `afterEach` block should be executed regardless of the test failing or not, this way we can avoid ùsing `try...finally` block in our test case